### PR TITLE
Automate test masters

### PR DESCRIPTION
**Pull request by chick**
Lots of changes but the main ones of interest at the changes to Readme and the
python scripts in `src/release_scripts`
I don't know if it's necessary to merge this yet, it might be better to perhaps to track this as a `dev` branch
while it is being fleshed out. I have successfully run `build_masters.py` (though I had to force running steps after step 6 because of expected problems with the test)

### CODE REVIEW

File: scripts/run_check_tests.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+source run_and_check.sh
+
+# check python virtual environment
+run_and_check 0.0 python $REPOTOOLS/src/utils/check_virtual_env.py
+
+run_and_check_neg 4.2 grep '\[error\]' run_tests_test.out
+
+run_and_check_neg 4.1 grep -n "Tests:.succeeded.*failed.[1-9]" run_tests_test.out
+
+# run_and_check 4.3 tail -100 run_tests_test.out
```
Comment (ucbjrl) [inline scripts/run_check_tests.sh]: missing newline at EOF?

File: scripts/setup_env.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+#
+# This script is designed to be sourced to create a proper environment for running release scripts
+#
+source ~/.virtualenvs/cit3/bin/activate
+export PYTHONPATH=/Users/chick/Adept/dev/release-generators/chisel-repo-tools/src
+export VERSIONING=/Users/chick/Adept/dev/release-generators/chisel-repo-tools/src/versioning/versioning.py
+sys.prefix == sys.base_prefix
```
Comment (ucbjrl) [inline scripts/setup_env.sh]: missing newline at EOF

File: src/repocontributors/repocon.py [hunk 3]
```diff
@@ -62,7 +62,7 @@ def __init__(self, area, reString):
         self.re = re.compile(self.reString)
 
 areaREs = [
-        areaRE('documentation', r'doc/'),
+        areaRE('documentation', r'docs/'),
```
Comment (ucbjrl) [inline src/repocontributors/repocon.py]: I don't know if this is useful in anything other than chisel2. If you aren't actually using it, I'd undo this edit just so it's obvious the file is way out of date.

**Review by ucbjrl — APPROVED** [review]
Great job Chick. I haven't looked at the new python code in any detail, but I like the overall approach.

File: src/versioning/versioning.py [hunk 4]
```diff
@@ -691,7 +691,9 @@ def main(argv=None) -> int: # IGNORE:C0111
     program_version = "v%s" % __version__
     program_build_date = str(__updated__)
     program_version_message = '%%(prog)s %s (%s)' % (program_version, program_build_date)
-    program_shortdesc = __import__('__main__').__doc__.split("\n")[1]
+    #program_shortdesc = __import__('__main__').__doc__.split("\n")[1]
```
Comment (chick) [inline src/versioning/versioning.py:694]: @ucbjrl I'm not sure what this is doing, my hack is just to get it to run right now.

**Comment by chick** [comment]
Current work has been switched to branch `dev` which will soon be PR's to master

**Pull request closed without merging**